### DISCUSSION TITLE
Make extensions discoverable by NFD

### DISCUSSION THREAD

**Discussion by SISheogorath** [siderolabs/talos#9070] [Ideas]
When nodes are configured differently, it can be hard to spot. E.g. if I have some nodes enabled with a different container runtime, like `gvisor` or `katacontainer`, it can be difficult to select these nodes.

For enabling the discovery of CPU features and alike kubernetes has [`node-feature-discovery`](https://github.com/kubernetes-sigs/node-feature-discovery). node-feature-discovery has a local feature, which the talos extension mechanism could hook into: https://github.com/kubernetes-sigs/node-feature-discovery/blob/bd8d74d6f2523b5cc866801f4e6bdb1d113ffd5d/docs/usage/customization-guide.md#local-feature-source

I first considered adding it to the extensions themselves but this will just be repeater over and over again. I have not found the right spot to implement it yet, but just providing the files in the mentioned directory, should enable any node-feature-discovery enabled cluster to select Talos extensions as node labels, while not causing any incompatibility with vanilla setups.

**Comment by smira** [comment]
Yes, this sounds like a good idea, we can allow system extensions to publish a file under `/etc/kubernetes/node-feature-discovery/features.d` to expose the labels for the node.

**Reply by smira** [reply]
On other hand, Talos could easily push to the machine labels itself without requiring yet another dependency.
